hosted/alibaba/helper/helper_cluster.go

```diff
@@ -0,0 +1,368 @@
+			cluster, err = client.Management.Cluster.ByID(cluster.ID)
+			Expect(err).To(BeNil())
+			return len(cluster.AliStatus.UpstreamSpec.NodePools)
+		}, tools.SetTimeout(helpers.Timeout), 10*time.Second).Should(BeNumerically("==", currentNodePoolNumber-1))
+		// The check for correct node pool names is complex after deletion, so we focus on the count.
+	}
+	return cluster, nil
+}
+
+// UpgradeClusterKubernetesVersion upgrades the control plane k8s version to the value defined by upgradeToVersion.
+// If checkClusterConfig is set to true, it will validate that the control plane has been upgraded successfully
+func UpgradeClusterKubernetesVersion(cluster *management.Cluster, upgradeToVersion string, client *rancher.Client, upgradeNodePool, wait, checkClusterConfig bool) (*management.Cluster, error) {
+	upgradedCluster := newClusterUpdatePayload(cluster)
+
+	// Ensure ImageID is populated for all node pools in the update payload
+	populateNodePoolImageIDs(upgradedCluster)
+
+	var upgradeCP bool
+	if cluster.AliConfig.KubernetesVersion != upgradeToVersion {
+		upgradedCluster.AliConfig.KubernetesVersion = upgradeToVersion
+		upgradeCP = true
+	}
+
+	ginkgo.GinkgoLogr.Info(fmt.Sprintf("Kubernetes version for cluster %s will be upgraded to %s", cluster.Name, upgradeToVersion))
+
+	var err error
+	cluster, err = client.Management.Cluster.Update(cluster, upgradedCluster)
+	if err != nil {
+		return nil, err
+	}
+
+	if wait {
+		err = clusters.WaitClusterToBeUpgraded(client, cluster.ID)
+		Expect(err).To(BeNil())
+	}
+
+	if checkClusterConfig {
+		if upgradeCP {
+			// Check if the desired config has been applied in Rancher
+			Eventually(func() bool {
+				ginkgo.GinkgoLogr.Info("Waiting for k8s upgrade to appear in AliStatus.UpstreamSpec & AliConfig ...")
+				cluster, err = client.Management.Cluster.ByID(cluster.ID)
+				Expect(err).To(BeNil())
+				return cluster.AliStatus.UpstreamSpec.KubernetesVersion == upgradeToVersion
+			}, tools.SetTimeout(helpers.Timeout), 30*time.Second).Should(BeTrue())
+		}
+	}
+	return cluster, nil
+}
+
+// ListALIAvailableVersions lists all available ALI Kubernetes versions
+// This is a separate static list maintained by hosted-providers-e2e, similar to the UI lists.
+func ListALIAllVersions(client *rancher.Client) (allVersions []string, err error) {
+	serverVersion, err := helpers.GetRancherServerVersion(client)
+	if err != nil {
+		// Log and continue with a safe default list; server-version is non-critical for ALI static list selection
+		ginkgo.GinkgoLogr.Info(fmt.Sprintf("[WARN] failed to read rancher server-version: %v; falling back to default ALI version list", err))
+	}
+
+	if serverVersion != "" && strings.Contains(serverVersion, "2.13") {
+		allVersions = []string{"1.33.3-aliyun.1", "1.32.7-aliyun.1", "1.32.6-aliyun.1"}
+	}
+
+	if serverVersion != "" && strings.Contains(serverVersion, "2.10") {
+		allVersions = []string{"1.31", "1.30", "1.29", "1.28"}
```

Comment:
This is not required, since Ali Baba is only supported on Rancher version >= 2.13.
Also, the k8s versions in the list does not match the aliyun supported versions.